### PR TITLE
Only include endpoint path in `<summary>`

### LINKED ISSUE #1352: Borked styling of list items under /register definition
**Link to problem area**: https://spec.matrix.org/v1.4/client-server-api/#post_matrixclientv3register

**Issue**
image

Was fine in 1.3, borked in 1.4 and 1.5.

### PR DESCRIPTION
**Pull request by DMRobertson**
As threatened on https://github.com/matrix-org/matrix-spec/issues/1358#issuecomment-1325220859 and https://github.com/matrix-org/matrix-spec/issues/1352#issuecomment-1439033334. (If nothing else, I seem to be consistent.)

Fixes #1352.

Preview: https://pr1446--matrix-spec-previews.netlify.app

### CODE REVIEW

**Comment by DMRobertson** [comment]
Seems to fix #1352. I can't remember the details but I vaguely remember from the last time I looked into this that there's something about being inside a `<summary>` that changes how lists are laid out.

**Comment by DMRobertson** [comment]
> I do seem to have lost the borders

This seems to be local to me and was because I had uncommitted changes in themes/docsy.

File: assets/scss/custom.scss [hunk 1]
```diff
@@ -288,16 +288,20 @@ footer {
...
   details {
     summary {
-      padding: .5rem 0;
-      p {
-        max-width: 80%;
+      h1 {
+        margin: 0;
+        /* Ensure the disclosure control is vertically centred with the header text. */
+        vertical-align: middle;
       }
```
Comment (richvdh) [inline assets/scss/custom.scss:300]: I guess this relies on the first line of the summary being an `<h1>`, but that's probably an assumption we can live with.

**Review by richvdh — APPROVED** [review]
thanks!

File: assets/scss/custom.scss (hunk 1 above)
Comment (DMRobertson) [reply assets/scss/custom.scss:300]: Could maybe do `summary > *` or `summary > :first-child` if this bites us in the future/

**Comment by DMRobertson** [comment]
Other details elements have this wierdly oversized summary too, e.g https://spec.matrix.org/v1.7/client-server-api/#mroomcanonical_alias

**Pull request merged**